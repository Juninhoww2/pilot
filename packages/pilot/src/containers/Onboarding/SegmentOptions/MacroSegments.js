import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import Card from '../Card'
import styles from './styles.css'

const MacroSegments = ({
  handleSubmit,
  images,
  notFoundText,
  options,
}) => (
  <div>
    <div className={styles.cards}>
      {options.map((option, index) => {
        const Image = images[index]
        const isLastCard = index === images.length - 1

        return (
          <Card
            className={classNames({
              [styles.lastCard]: isLastCard,
            })}
            key={option.category}
            onSubmit={() => handleSubmit(option.category)}
          >
            <div className={styles.cardContent}>
              <Image width={32} height={32} />
              <p className={styles.cardLabel}>
                {option.label}
              </p>
            </div>
          </Card>
        )
      })}
    </div>

    <div className={styles.notFound}>
      <button
        onClick={() => handleSubmit('not_found')}
        role="link"
        type="button"
      >
        {notFoundText}
      </button>
    </div>
  </div>
)

MacroSegments.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  images: PropTypes.arrayOf(PropTypes.func),
  notFoundText: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      category: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired,
    })
  ).isRequired,
}

MacroSegments.defaultProps = {
  images: [],
}

export default MacroSegments
