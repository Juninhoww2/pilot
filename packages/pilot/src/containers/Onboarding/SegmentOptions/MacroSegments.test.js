import React from 'react'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import MacroSegments from './MacroSegments'

test('MacroSegments should render', () => {
  const notFoundText = 'Não encontrei o segmento do meu negócio'
  const options = [{
    category: 'food',
    label: 'Alimentos',
  }, {
    category: 'others',
    label: 'Outros',
  }]
  const handleSubmit = jest.fn()
  const svg = () => <svg />

  const { container, getByText } = render(
    <MacroSegments
      notFoundText={notFoundText}
      images={[svg, svg]}
      options={options}
      handleSubmit={handleSubmit}
    />
  )

  userEvent.click(getByText('Alimentos'))
  expect(handleSubmit).toHaveBeenCalledTimes(1)

  userEvent.click(getByText('Outros'))
  expect(handleSubmit).toHaveBeenCalledTimes(2)

  userEvent.click(getByText(notFoundText))
  expect(handleSubmit).toHaveBeenCalledTimes(3)

  expect(container).toMatchSnapshot()
})
