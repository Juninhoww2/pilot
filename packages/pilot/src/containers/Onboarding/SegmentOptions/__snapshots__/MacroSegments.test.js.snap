// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MacroSegments should render 1`] = `
<div>
  <div>
    <div
      class="cards"
    >
      <div
        class="clickableCard"
        role="button"
        tabindex="0"
      >
        <div
          class="card"
        >
          <div
            class="cardContent"
          >
            <svg />
            <p
              class="cardLabel"
            >
              Alimentos
            </p>
          </div>
        </div>
      </div>
      <div
        class="clickableCard"
        role="button"
        tabindex="0"
      >
        <div
          class="card lastCard"
        >
          <div
            class="cardContent"
          >
            <svg />
            <p
              class="cardLabel"
            >
              Outros
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="notFound"
    >
      <button
        role="link"
        type="button"
      >
        Não encontrei o segmento do meu negócio
      </button>
    </div>
  </div>
</div>
`;
